R/compiled.R

```diff
@@ -1,17 +1,17 @@
 # this does not handle LCOV_EXCL_START ect.
 parse_gcov <- function(file, package_path = "") {
-  if (!file.exists(file)) {
+  if (!source_file_exists(file)) {
     return(NULL)
   }
 
-  lines <- readLines(file)
+  lines <- read_lines_impl(file)
   source_file <- rex::re_matches(lines[1], rex::rex("Source:", capture(name = "source", anything)))$source
 
   # retrieve full path to the source files
   source_file <- normalize_path(source_file)
 
   # If the source file does not start with the package path or does not exist ignore it.
-  if (!file.exists(source_file) || !grepl(rex::rex(start, rex::regex(paste0(rex::escape(package_path), collapse = "|"))), source_file)) {
+  if (!source_file_exists(source_file) || !grepl(rex::rex(start, rex::regex(paste0(rex::escape(package_path), collapse = "|"))), source_file)) {
     return(NULL)
```

Comment:
These changes are required to remove the mockery dependency. I kept the commit self-contained.

But basically, with `mockery::stub()`, we can mock a function contained to a function. For example, mocking `file.exists()` only inside `parse_gcov()`. However, `testthat::local_mocked_bindings()` doesn't provide this ability.

The workaround I found is to create a wrapper function for `file.exists()` (in this case `source_file_exists()`.) that I will only use inside `parse_gcov()`. This way, I can mock `source_file_exists()` to only mock `file.exists()` inside `parse_gcov()`.